### PR TITLE
Jetpack: redirect users to "My Jetpack" upon plugin activation

### PR DESCRIPTION
**Pull request by sergeymitr**
## Proposed changes:
* Redirect users to "My Jetpack" upon plugin activation.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
p1HpG7-qWK-p2

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Go to "Plugins" and deactivate Jetpack and all standalone plugins.
2. Activate Jetpack.
3. Confirm you got redirected to the "My Jetpack" page.
4. If you're using a new site without any standalone plugins installed, "My Jetpack" should display the "Welcome to Jetpack!" banner.

### CODE REVIEW

File: projects/plugins/jetpack/class.jetpack.php [hunk 1]
```diff
@@ -3332,6 +3332,9 @@ public function admin_init() {
 		/** Already documented in automattic/jetpack-connection::src/class-client.php */
 		$client_verify_ssl_certs = apply_filters( 'jetpack_client_verify_ssl_certs', false );
 
+		// Run post-activation actions if needed.
+		$this->plugin_post_activation();
```
Comment (jboland88) [inline projects/plugins/jetpack/class.jetpack.php:3336]: do we need to wait for the admin to load in order to call `wp_safe_redirect`? 
Would it make sense to just redirect in the plugin activation hook?

**Review by jboland88 — APPROVED** [review]
I wonder if we could move the redirect into the activation hook - but I don't think that's a blocker.

Tested well for me.

File: projects/plugins/jetpack/class.jetpack.php (hunk 1 above)
Comment (sergeymitr) [reply projects/plugins/jetpack/class.jetpack.php:3336]: Discussed further in Slack: p1707156725034619-slack-C01264051NE

**Comment by samiff** [comment]
cherry-picked to `weekly/branch-2024-02-05` in 30ae7ebf8166121e62f78cfc56b69a9653c7ee9a for the 13.1 release.

**Pull request merged**